### ISSUE TITLE
detect 'bad' function names.

### ISSUE THREAD

**Issue by bittorf** [koalaman/shellcheck#529]
according to http://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap08.html
a valid function name is like:
```
"consist solely of uppercase letters, digits, and the
'_' (underscore) from the characters defined in Portable
Character Set and do not begin with a digit"
```

the following does not throw an error whith shellsheck, but it should detect it:

```shell
#!/bin/sh
x-y()
{
  :
}

x-y
```
the same for variables names, but i have not tested it yet.
even not all shells complain about this with 'sh -n $file'.

**Comment by koalaman** [comment]
Nice! Added in 7415c9d.
